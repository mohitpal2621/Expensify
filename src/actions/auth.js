import { auth, googleAuthProvider } from "../firebase/firebase";
import { signInWithPopup, signOut } from "firebase/auth";

export const login = (uid) => ({
    type: 'LOGIN',
    uid
});

export const logout = () => ({
    type: 'LOGOUT',
})

export const startLogin = () => {
    return async () => {
        await signInWithPopup(auth, googleAuthProvider);
    };
};

export const startLogout = () => {
    return async () => {
        await signOut(auth);
    };
};
