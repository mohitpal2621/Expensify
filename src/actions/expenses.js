import { v4 as uuid } from "uuid";

// ADD_EXPENSE action generator
export const addExpense = ({ description='', note='', amount=0, createdAt=0 }) => ({
    type: 'ADDEXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

// REMOVE_EXPENSE action generator
export const removeExpense = (id) => ({
    type: 'REMOVEEXPENSE',
    id
});

// EDIT_EXPENSE action generator
export const editExpense = (id, updates) => ({
        type: 'EDITEXPENSE',
        id,
        updates
});