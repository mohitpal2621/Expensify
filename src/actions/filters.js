// SETTEXTFILTER action generator
export const setTextFilter = (text = '') => ({
    type: 'SETTEXT',
    text
});

// SORTBYAMOUNT action generator
export const sortByAmount = () => ({type: 'SORTBYAMOUNT'});

// SORTBYDATE action generator
export const sortByDate = () => ({type: 'SORTBYDATE'});

// SETSTARTDATE action generator
export const setStartDate = (stDate) => ({
    type: 'SETSTARTDATE',
    stDate
});

// SETENDDATE action generator
export const setEndDate = (edDate) => ({
    type: 'SETENDDATE',
    edDate
});
