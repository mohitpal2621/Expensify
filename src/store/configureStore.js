import { combineReducers, configureStore } from "@reduxjs/toolkit";
import expenseReducer from '../reducers/expenseReducer';
import filtersReducer from '../reducers/filterReducer';
import authReducer from "../reducers/authReducer";

const rootreducer = combineReducers({
    expenses: expenseReducer,
    filters: filtersReducer,
    auth: authReducer
});

export default () => {
    const store = configureStore({
        reducer: rootreducer
    });

    return store;
};
