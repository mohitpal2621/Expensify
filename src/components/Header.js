import React from "react";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";
import { startLogout } from "../actions/auth";


const Header = (props) => {
    return (
        <header>
            <h1>Expensify</h1>          
            <nav id="navbars">
                <NavLink to="/dashboard" className={({ isActive }) => {
                    return isActive ? "active" : "inactive";
                }}>
                    Home
                </NavLink>
                <NavLink to="/create" className={({ isActive }) => {
                    return isActive ? "active" : "inactive";
                }}>
                    Create
                </NavLink>
            </nav>
            <button onClick={props.startLogout}>Log Out</button>
        </header>
    );
};

const mapDispatchToProps = (dispatch) => ({
    startLogout: async () => await dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);
