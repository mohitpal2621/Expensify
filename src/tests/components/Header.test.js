import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import Header from '../../components/Header';
import { Provider } from 'react-redux';
import configureStore from '../../store/configureStore';
import { startLogout } from '../../actions/auth';

let store;

beforeEach(() => {
    store = configureStore(); // Use your actual store configuration
});

jest.mock('../../actions/auth.js', () => ({
    startLogout: jest.fn(),
}));

// jest.mock('firebase/auth', () => ({
//     getAuth: jest.fn(() => ({
//       signOut: jest.fn(),
//     })),
//     GoogleAuthProvider: jest.fn().mockImplementation(() => ({
//         setCustomParameters: jest.fn(),
//     })),
// }));

test("Should render Header correctly", () => {
    const { container } = render(
        <Provider store={store}>
            <MemoryRouter>
                <Header startLogout={() => {}}/>
            </MemoryRouter>
        </Provider>
    );
    expect(container).toMatchSnapshot();
});

// test("Should call startLogout on button click", async () => {
//     const startLogoutMock = jest.fn(); // Mock the startLogout action creator

//     const { getByText } =  render(
//         <Provider store={store}>
//             <MemoryRouter>
//                 <Header startLogout={startLogoutMock} />
//             </MemoryRouter>
//         </Provider>
//     );

//     fireEvent.click(getByText("Log Out")); // Simulate button click

//     await expect(startLogoutMock).toHaveBeenCalled(); // Assert that startLogout was called
// });