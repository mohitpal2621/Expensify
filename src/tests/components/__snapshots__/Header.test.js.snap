// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<div>
  <header>
    <h1>
      Expensify
    </h1>
    <nav
      id="navbars"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        class="inactive"
        href="/create"
      >
        Create
      </a>
    </nav>
    <button>
      Log Out
    </button>
  </header>
</div>
`;
